src/bindings/jna/libelektra4j/src/main/java/org/libelektra/exception/mapper/ExceptionMapperService.java

```diff
@@ -0,0 +1,49 @@
+package org.libelektra.exception.mapper;
+
+import org.libelektra.Key;
+import org.libelektra.exception.*;
+
+public class ExceptionMapperService {
+
+	public static KDBException getMappedException(Key k) {
+
+		// At least one error or warning should be there
+		if (k.getMeta("error/number").isNull() && k.getMeta("warnings/#00/number").isNull()) {
+			Key temporaryError = Key.create("user/temporary/errorkey");
+			temporaryError.setMeta("error/number", InternalException.errorNumber());
+			temporaryError.setMeta("error/reason", "Sorry, Elektra returned an error/warning but no " +
+					"information along it, please report this incident at https://issues.libelektra.org/");
+			return new InternalException(temporaryError);
```

Comment:
Just a question here: If you return this exception and later someone calls `exception.getModule()`, would that fail? Because the `error/module` key is not set here and `getModule` doesn't do any checks and doesn't throw exceptions either: 
```java
public String getModule() [ getModule()]  {
		return errorKey.getMeta("error/module").getString();
	}
```

I'm implementing the kdb errors for the Rust binding atm, so I'm looking at this PR for some inspiration and noticed that. So to fully handle the case where an error key returned from a kdb function is empty, shouldn't all the `error/*` metakeys be set?

Reply:
Yes, I also found the code weird, see my comments two lines above.

From my perspective, the `kdb*` calls must properly set the errors. It would be wrong to fix problems from the core in every binding separately.

> So to fully handle the case where an error key returned from a kdb function is empty, shouldn't all the error/* metakeys be set?

This case does not need to be handled, as it would be a bug in the core and needs to be fixed there. And it is actually quite simple to fix all such errors: for every `return -1` in the `kdb` functions, we simply need to check if an error was correctly set, and set some error (similar to the code above) if it was not set.